### PR TITLE
NUTCH-3078 Unlock database when Injector finishes - regardless of result

### PR DESCRIPTION
**Pull request by HiranChaudhuri**

### CODE REVIEW

File: src/java/org/apache/nutch/crawl/Injector.java [hunk 1]
```diff
@@ -402,40 +402,38 @@ public void inject(Path crawlDb, Path urlDir, boolean overwrite,
 
     // lock an existing crawldb to prevent multiple simultaneous updates
     Path lock = CrawlDb.lock(conf, crawlDb, false);
```
Comment (sebastian-nagel) [inline src/java/org/apache/nutch/crawl/Injector.java:404]: See the discussion in NUTH-3078: moving the creating of the lockfile down just before the job is run would avoid that a failure during job configuration leads to a lost lockfile.

**Review by sebastian-nagel — CHANGES_REQUESTED** [review]

File: src/java/org/apache/nutch/crawl/Injector.java (hunk 1 above)
Comment (sebastian-nagel) [reply src/java/org/apache/nutch/crawl/Injector.java:440]: Great!

**Review by sebastian-nagel — APPROVED** [review]
Thanks,  @HiranChaudhuri!

I've also verified that this solution fixes the issue with the lockfile if the seed URL directory does not exists.

**Review by lewismc — APPROVED** [review]
+1 nice PR @HiranChaudhuri

**Pull request merged**